app/models/grade_entry_student.rb

```diff
@@ -177,7 +177,7 @@ def self.get_total_grades(ges_ids)
     grades.map do |k, v|
       ges_grades = v.map(&:last)
       ges_grades = ges_grades.all?(&:nil?) ? nil : ges_grades.compact.sum
-      [k, ges_grades]
+      [k, ges_grades.nil? ? nil : [ges_grades, 0].max]
```

Comment:
I'd recommend doing the `max` on the previous line, since that already differentiates between the `nil` and non-`nil` cases